internal/controller/dnsrecord_controller.go

```diff
@@ -546,7 +550,7 @@ func setStatusConditions(record *v1alpha1.DNSRecord, hadChanges bool, notHealthy
 	}
 
 	// if we haven't published because of the health failure, we won't have changes but the spec endpoints will be empty
-	if record.Status.Endpoints == nil || len(record.Status.Endpoints) == 0 {
+	if len(record.Status.Endpoints) == 0 {
```

Comment:
Why was this removed? Was this a required change for coredns?

Reply:
hmm not sure now. May have been accidental

Reply:
ahh actually I think the nil check is not needed as it will default to 0 if not set

Reply:
```
type val struct{}

func main() {

	var empty []*val
	fmt.Println(len(empty))
}
```